List.lua

```diff
@@ -155,49 +155,54 @@ function lls.List:Print()
 
     -- Append last line
     if ZoUTF8StringLength(summary) > ZoUTF8StringLength(self.prefix) then
-        table.insert(lines, summary)
+        tableInsert(lines, summary)
     end
 
     -- Print to chat
-    for _, line in ipairs(lines) do
+    for i, line in ipairs(lines) do
         self.chat:Print(self.prefix .. line .. self.suffix)
+        lines[i] = nil -- to allow collect garbage
     end
 
     self:Reset()
 end
 
-function lls.List:Reset()
-    self.itemList = {}
-    self.itemKeys = {}
-    self.currencyList = {}
-    self.currencyKeys = {}
+function List:Reset()
+    ZO_ClearTable(self.itemList)
+    ZO_ClearTable(self.itemKeys)
+    ZO_ClearTable(self.currencyList)
+    ZO_ClearTable(self.currencyKeys)
```

Comment:
improvement for garbagecollector

Reply:
1. self.itemKeys and self.currencyKeys only hold strings, so they don't require any clearing.
2. self.itemList and self.currencyList can just be replaced with {} if they are marked as weak tables with a setmetatable(varname, { __mode="v" }). See https://www.lua.org/pil/17.html.  This should perform better at runtime than iterating over the whole table and clearing.